tests/scripts/all.sh

```diff
@@ -1151,22 +1118,7 @@ component_build_arm_none_eabi_gcc_no_64bit_multiplication () {
 
 component_build_armcc () {
     msg "build: ARM Compiler 5, make"
-    scripts/config.pl full
-    scripts/config.pl unset MBEDTLS_NET_C
-    scripts/config.pl unset MBEDTLS_TIMING_C
-    scripts/config.pl unset MBEDTLS_FS_IO
-    scripts/config.pl unset MBEDTLS_ENTROPY_NV_SEED
-    scripts/config.pl unset MBEDTLS_HAVE_TIME
-    scripts/config.pl unset MBEDTLS_HAVE_TIME_DATE
```

Comment:
Does baremetal unset `MBEDTLS_HAVE_TIME`? If so, is it really equivalent for tests like `component_build_arm_none_eabi_gcc_no_64bit_multiplication()` that didn't unset `MBEDTLS_HAVE_TIME` previously? If not, should we keep unsetting `MBEDTLS_HAVE_TIME` (and other time-related configurations) for this `component_build_armcc()` test?

Reply:
Yes, `scripts/config.pl baremetal` unsets `HAVE_TIME` (and `HAVE_TIME_DATE` as well), so it doesn't change the `build_armcc` component in that respect.

Regarding the test for no 64-bit multiplication, I think we can agree it's quite unlikely that calling `time()` or `gmtime()` (which is what `HAVE_TIME` and `HAVE_TIME_DATE` are guarding) in SSL and X.509 is unlikely to affect whether we use 64-bit mulliplication in crypto modules, so whether or not we enable those calls shouldn't matter for this test.

I'd argue that the fact that these calls are available (but by the way, unlikely to actually work at runtime) when building with arm-none-eabi-gcc, while they are not with other embedded toolchains such as armcc, is an accidental detail, and that it's better to just disable calls to libc time functions whenever we build for an embedded target, unless we specifically intend to test those functions on that particular target (which we're not doing anywhere in `all.sh` so far). (If we didn't I'd even argue for doing `scripts/config.pl baremetal && scripts/config.pl set MBEDTLS_HAVE_TIME` in order to make the difference explicit.)